go-controller/pkg/clustermanager/zone_cluster_controller.go

```diff
@@ -162,6 +182,88 @@ func (zcc *zoneClusterController) syncNodeIds(nodes []interface{}) error {
 	return nil
 }
 
+func (zcc *zoneClusterController) syncZoneJoinSubnet(node *corev1.Node) error {
```

Comment:
at startup I get it that we need a one time thing to sync this across ALL nodes, but isn't it better if we moved this to `syncNodes` ? do we expect this to get out of sync everytime a node is updated? we generally don't call sync functions from add/update always.

 reason I am asking is because components across OCP might change node annotations for different reasons and even annotations we don't care about! all that will generate unnecessary syncs? In KIND this won't be visible since are are smart in processing an update only if annotation on node changes. Maybe update your `AreResourcesEqual` to include annotations we care about like said here: https://github.com/ovn-org/ovn-kubernetes/pull/3169#discussion_r1151815513